core/shared/src/main/scala/io/chrisdavenport/log4cats/StructuredLogger.scala

```diff
@@ -21,49 +21,54 @@ object StructuredLogger {
   def apply[F[_]](implicit ev: StructuredLogger[F]): StructuredLogger[F] = ev
 
   def withContext[F[_]](sl: StructuredLogger[F])(ctx: Map[String, String]): StructuredLogger[F] =
-    new ExtraContextStructuredLogger[F](sl)(ctx)
+    new ModifiedContextStructuredLogger[F](sl)(_ ++ ctx)
 
-  private class ExtraContextStructuredLogger[F[_]](sl: StructuredLogger[F])(
-      ctx: Map[String, String]
+  def withModifiedContext[F[_]](
+      sl: StructuredLogger[F]
+  )(modifyCtx: Endo[Map[String, String]]): StructuredLogger[F] =
+    new ModifiedContextStructuredLogger[F](sl)(modifyCtx)
+
+  private class ModifiedContextStructuredLogger[F[_]](sl: StructuredLogger[F])(
+      modify: Endo[Map[String, String]]
```

Comment:
While `Endo` is really cool, I think it might trip up more people than necessary. I think the classic function syntax would be more approachable.